### PR TITLE
Add language definition for robots.txt

### PR DESCRIPTION
**Pull request by thomasleveil**

### CODE REVIEW

File: docs/css-classes-reference.rst [hunk 1]
```diff
@@ -392,6 +392,8 @@ Language names and aliases
 +-------------------------+---------------------------------------------------+
 | Roboconf                | graph, instances                                  |
 +-------------------------+---------------------------------------------------+
+| robots.txt              | robotstxt, rotbots.txt                            |
```
Comment (yyyc514) [inline docs/css-classes-reference.rst]: Typo?

File: src/languages/robotstxt.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    keywords: {
+      section: 'user-agent',
+      built_in: 'allow disallow',
+      keyword: 'crawl-delay sitemap'
+    },
+    contains: [
+      HASH_COMMENT_MODE,
+      hljs.NUMBER_MODE,
+      {
+        className: 'regex',
```
Comment (yyyc514) [inline src/languages/robotstxt.js]: Why not string?  Isn't * the only special character allowed and these aren't regexes?

**Comment by yyyc514** [comment]
I think now we're wanting new languages to be in separate repos going forward.  Are you wanting to be the maintainer for Robots.txt going forward?  

If so it might be possible to create a repository in the highlightjs organization for it to live.

**Comment by thomasleveil** [comment]
> Should we not also match the urls?

I tried, but having not much experience I had difficulties making it work properly. I felt this was enough for my use case

**Comment by thomasleveil** [comment]
> I think now we're wanting new languages to be in separate repos going forward. Are you wanting to be the maintainer for Robots.txt going forward?

I had the need for this feature and felt it may help others. However I won't be able to maintain this.

**Comment by yyyc514** [comment]
@egor-rogov I think I could probably support this one over time (robots.txt is a pretty static thing, lol).  Is there a process to go about requesting an official language repository?

**Comment by thomasleveil** [comment]
I have no clue. This is my first time contributing on this project

**Comment by yyyc514** [comment]
I was directing that to Egor. :-)

**Comment by egor-rogov** [comment]
I'll leave it to @marcoscaceres to decide. I agree it's pretty static and quite simple, but...

**Comment by yyyc514** [comment]
I wasn't recommending we add it to core, I was volunteering to be the maintainer of the official repository for the syntax.

**Comment by egor-rogov** [comment]
Ah, okay then. I've just created a repo: https://github.com/highlightjs/highlightjs-robots-txt
Copy language definition to it and let's see if this works (I've never done this before).

**Comment by yyyc514** [comment]
> Copy language definition to it and let's see if this works (I've never done this before).

What are you expecting to work or not work? :-)  You still have to link to the language just like you would any other...

**Comment by yyyc514** [comment]
@thomasleveil You fine with MIT license?

**Comment by yyyc514** [comment]
@thomasleveil I made the commit but gave you the authorship, you're mentioned in README also.

**Comment by egor-rogov** [comment]
@yyyc514 
> What are you expecting to work or not work? :-)

I mean, privileges to use the repo etc. I see it's OK.

**Comment by yyyc514** [comment]
> I mean, privileges to use the repo etc. I see it's OK.

Ah yes, though I was confused because I only glanced at the email then tried to push but you first have to ACCEPT the invitation.  After I did that, all was well. :-)

**Comment by thomasleveil** [comment]
> @thomasleveil You fine with MIT license?

@yyyc514 I'm fine with  it

**Comment by yyyc514** [comment]
Closing this as we now have the new repo. :)

**Pull request closed without merging**